cairo/ethereum/cancun/vm.cairo

```diff
@@ -88,6 +105,311 @@ struct MessageStruct {
     parent_evm: OptionalEvm,
 }
 
+func incorporate_child_on_success{range_check_ptr, poseidon_ptr: PoseidonBuiltin*, evm: Evm}(
+    child_evm: Evm
+) {
+    alloc_locals;
+
+    let fp_and_pc = get_fp_and_pc();
+    local __fp__: felt* = fp_and_pc.fp_val;
+
+    let new_gas_left = Uint(evm.value.gas_left.value + child_evm.value.gas_left.value);
+
+    let dst = evm.value.logs.value.data + evm.value.logs.value.len;
+    let src = child_evm.value.logs.value.data;
+    let len = child_evm.value.logs.value.len;
+    memcpy(dst, src, len);
+    tempvar new_logs = TupleLog(
+        new TupleLogStruct(data=evm.value.logs.value.data, len=evm.value.logs.value.len + len)
+    );
+
+    let new_refund_counter = evm.value.refund_counter + child_evm.value.refund_counter;
```

Comment:
This could overflow. Add a comment why this cannot or add a check to ensure it does not.

Reply:
Afaik gas left cannot overflow since the 'gas_left' from the child is taken as a fraction of the parent's gas left, such that for any sub-call in the stack, child_gas_left is always smaller than the parent_gas_left

Reply:
I agree but the more we advance the more we need to think about auditors and external readers.
Context is really important and details can slip through. 
By design, all additions can overflow in Cairo0. When there is no overflow checks we should justify it imo.